tests/trainer/test_trainer_cli.py

```diff
@@ -71,11 +72,34 @@ def test_add_argparse_args_redefined(cli_args):
 def test_add_argparse_via_argument_group(cli_args):
     """Simple test ensuring that passing an argument group still works"""
     parser = ArgumentParser(add_help=False)
-    parser = Trainer.add_argparse_args(parser.add_argument_group(title="pl.Trainer args"))
+    parser = Trainer.add_argparse_args(
```

Comment:
Can we automate this in Trainer.add_argparse_args ? I don't except users to know about this.

Reply:
I wouldn't make the choice to use `ArgumentGroup` for the user.

Reply:
Agreed that it should be up to the user.
(Personally, I would prefer it, but there may be weird edge cases if this is done by default at this point... mebbe good for future PR?)

Reply:
Maybe I spoke too soon. After latest push, maybe it would be best to just make this the default?

Then all the extra stuff, warnings, etc. all disappear. The only warning is to not do `cls.add_argparse_args(parser.add_argument_group("..."))`, and instead just the normal way.

Thoughts @carmocca / @Borda? It would simplify user interaction TBH.

Reply:
Well that'd open the door for somebody to open another PR in a few months so he doesn't have to use `ArgumentGroup`. I would like to avoid that

If you can add a flag to disable `ArgumentGroup`, then that's okay

Reply:
Done. Added explicit `use_argument_group=True` arg. (avoiding double-neg)